entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -717,21 +717,30 @@ class EntityService(
         neo4jRepository.deleteEntity(entityId)
 
     @Transactional
-    fun deleteEntityAttribute(entityId: String, attributeName: String, contextLink: String): Boolean {
+    fun deleteEntityAttribute(entityId: String, attributeName: String, datasetId: URI?, deleteAll: Boolean, contextLink: String): Boolean {
         val expandedAttributeName = expandJsonLdKey(attributeName, contextLink)!!
 
-        if (neo4jRepository.hasPropertyOfName(EntitySubjectNode(entityId), expandedAttributeName))
-            return neo4jRepository.deleteEntityProperty(entityId, expandedAttributeName) >= 1
-        else if (neo4jRepository.hasRelationshipOfType(
-            EntitySubjectNode(entityId),
-            expandedAttributeName.toRelationshipTypeName())
-        )
-            return neo4jRepository.deleteEntityRelationship(
-                entityId,
-                expandedAttributeName.toRelationshipTypeName()
-            ) >= 1
-
-        throw ResourceNotFoundException("Attribute $attributeName not found in entity $entityId")
+        if (deleteAll) {

```

Comment:
i'm not a big fan of flags for handling that kind of behaviour... maybe decoupling in methods and regroup equivalent code in private method could help for future changes

Reply:
`deleteAll` will still be a flag ... not sure of what you will gain here?

Reply:
I don't know the "real" gain, but in my mind is clearer to have 2 methods ```deleteEntityAttribute``` and ```deleteAllEntityAttributes```.. it removes the ```if-else``` part which reduces a bit the complexity.

Reply:
And deleteAll could have a reduced number of parameters as datasetId is not taken into account.

Reply:
OK, I see what you meant. Looks good.